eZ/Publish/Core/Persistence/Legacy/Content/Location/Handler.php

```diff
@@ -332,24 +332,30 @@ public function copySubtree($sourceId, $destinationParentId, $newOwnerId = null)
         }
 
         $destinationParentSectionId = $this->getSectionId($destinationParentId);
-        $this->updateSubtreeSectionIfNecessary($copiedSubtreeRootLocation, $destinationParentSectionId);
+
+        // potentially it may occur that the destination Location doesn't have a section (like in Location ID = 1),
+        // therefore assigning any or empty section will be invalid here
+        if ($destinationParentSectionId) {
+            $this->updateSubtreeSectionIfNecessary($copiedSubtreeRootLocation, $destinationParentSectionId);
+        }
 
         return $copiedSubtreeRootLocation;
     }
 
     /**
      * Retrieves section ID of the location's content.
-     *
-     * @param int $locationId
-     *
-     * @return int
      */
-    private function getSectionId($locationId)
+    private function getSectionId(int $locationId): ?int
```

Comment:
We should not add those type hints. As `$destinationParentId`, which is passed from 
```
    /**
     * @param mixed $sourceId
     * @param mixed $destinationParentId
     *
     * @return bool
     */
    public function move($sourceId, $destinationParentId)
```
is declared as mixed and it could introduce BC brake